common/flatpak-transaction.c

```diff
@@ -2382,14 +2470,71 @@ op_get_runtime_ref (FlatpakTransactionOperation *op)
   return decomposed;
 }
 
+static FlatpakDecomposed *
+op_get_sdk_ref (FlatpakTransactionOperation *op)
+{
+  g_autofree char *sdk_pref = NULL;
+  FlatpakDecomposed *decomposed;
+
+  if (!op->resolved_metakey || !flatpak_decomposed_is_app (op->ref))
+    return NULL;
+
+  sdk_pref = g_key_file_get_string (op->resolved_metakey, "Application", "sdk", NULL);
+  if (sdk_pref == NULL)
+    return NULL;
+
+  decomposed = flatpak_decomposed_new_from_pref (FLATPAK_KINDS_RUNTIME, sdk_pref, NULL);
+  if (decomposed == NULL)
+    g_debug ("Invalid runtime ref %s in metadata", sdk_pref);
+
+  return decomposed;
+}
+
+static gboolean
+add_new_dep_op (FlatpakTransaction           *self,
+                FlatpakTransactionOperation  *op,
+                FlatpakDecomposed            *dep_ref,
+                FlatpakTransactionOperation **dep_op,
+                GError                      **error)
+{
+  FlatpakTransactionPrivate *priv = flatpak_transaction_get_instance_private (self);
+  g_autofree char *dep_remote = NULL;
+
+  if (!ref_is_installed (self, dep_ref))
+    {
+      dep_remote = find_runtime_remote (self, op->ref, op->remote, dep_ref, op->kind, NULL, error);
+      if (dep_remote == NULL)
+        return FALSE;
+
+      *dep_op = flatpak_transaction_add_op (self, dep_remote, dep_ref, NULL, NULL, NULL, NULL,
+                                            FLATPAK_TRANSACTION_OPERATION_INSTALL_OR_UPDATE, FALSE, error);
+      if (*dep_op == NULL)
+        return FALSE;
+    }
+  else
+    {
+      /* Update if in same dir */
+      if (dir_ref_is_installed (priv->dir, dep_ref, &dep_remote, NULL))
+        {
+          g_debug ("Updating dependency %s", flatpak_decomposed_get_pref (dep_ref));
```

Comment:
Maybe mention what it's a dependency of, and add a `g_debug()` for the install case above? Otherwise this just duplicates the debug statement in `flatpak_transaction_add_op`